src/odemis/driver/orsay.py

```diff
@@ -2471,51 +2475,50 @@ def _updateImageFormat(self, parameter=None, attr_name="Actual"):
             parameter = self._ionColumn.ImageSize
         if attr_name != "Actual":
             return
+
         state = self._ionColumn.ImageSize.Actual
-        logging.debug("Image format is: %s. Updating translation and resolution and their ranges accordingly." % state)
+        logging.debug("Received ImageSize: %s.", state)
         new_value = tuple(map(int, state.split(" ")))
-        self.imageFormat._value = new_value  # to not call the setter
-        self.imageFormat.notify(new_value)
+
+        if new_value[0] != new_value[1]:
+            # For the scale, we only support square image format
+            logging.warning("Image format received is not square: %s", new_value)
+
+        if self.imageFormat._value != new_value:
+            self.imageFormat._value = new_value  # to not call the setter
+            self.imageFormat.notify(new_value)
 
     def _clip_and_set_image_area(self, target_resolution, target_translation):
```

Comment:
I'd rather not, because it's a private function. I prefer doing just "black box" testing, by only using the standard interface. So it's indirectly tested when testing the .translation and .resolution VAs, but not explicitly tested.